src/components/events-table.js

```diff
@@ -59,25 +64,145 @@ function inputDataRow(processEvent) {
 }
 
 const EventsTable = ({ taskData }) => {
-    let processEvents = taskData === null ? [] : taskData.processEvents;
+    const [anchorLevelEl, setAnchorLevelEl] = React.useState(null);
+    const [anchorLogEl, setAnchorLogEl] = React.useState(null);
+
+    const [processEvents] = React.useState(taskData.processEvents);
+    const [filtredProcessEvents, setFiltredProcessEvents] = React.useState(
+        taskData.processEvents
+    );
+
+    const [levelFilter, setLevelFilter] = React.useState('');
+    const [logFilter, setLogFilter] = React.useState('');
+    const handleLevelMenuClick = (event) => {
+        setAnchorLevelEl(event.currentTarget);
+    };
+
+    const handleLevelChange = (event) => {
+        setLevelFilter(event.currentTarget.value.toUpperCase());
+        filterProcessEvent(event.currentTarget.value.toUpperCase(), logFilter);
+    };
+
+    const handleLogMenuClick = (event) => {
+        setAnchorLogEl(event.currentTarget);
+    };
+    const handleLogChange = (event) => {
+        setLogFilter(event.currentTarget.value.toUpperCase());
+        filterProcessEvent(
+            levelFilter,
+            event.currentTarget.value.toUpperCase()
+        );
+    };
+
+    const filterProcessEvent = (currentEventFIlter, currentLogFilter) => {
+        let filtered;
+        filtered = processEvents.filter(
+            (event) =>
+                event.level.includes(currentEventFIlter) &&
+                event.message.toUpperCase().includes(currentLogFilter)
```

Comment:
It is a bit confusing what is in uppercase ad what is not. I would rather call the uppercase only in the comparison method, so you're sure you're comparing the good elements. This way you can store the value without uppercase for logFilter and levelFilter and you can pass these values directly to the method

Reply:
In fact everything is in uppercase. I will modify that point in order to clearly show what is compared